rnd/autogpt_server/autogpt_server/data/block.py

```diff
@@ -0,0 +1,130 @@
+import inspect
+import json
+
+from abc import ABC, abstractmethod
+from prisma.models import AgentBlock
+from pydantic import BaseModel
+from typing import ClassVar
+
+
+class Block(ABC, BaseModel):
+    @property
+    @abstractmethod
+    def input_schema(self) -> dict[str, str]:
+        """
+        The schema for the block input data. The keys are the names of the input,
+        and the values are the types of the input.
+        Example:
+        {
+            "system_prompt": "str",
+            "user_prompt": "str",
+            "max_tokens": "int",
+        }
+        """
+        pass
```

Comment:
Wouldn't is be better to use `jsonschema` or something? Because with`dict[str, str]` we cannot make type-safe nested schemas.

Reply:
Considering the input will be connected to the output of another block, I think we should avoid nested schema at all costs and make each input/output as granular as possible. Otherwise, it's going to be impossible to match the custom class of the output of one box as the input custom class of another box.

We still can, though, set the type on one of the input/output to the common/predefined types and we will simply use the type name as its type (as opposed to the proto/jsonschema structure)

Reply:
I updated to allow both, so by default you can simply provide a string, it will be primitive, but you can also provide more complicated jsonschema if needed.